File: src/main/java/fi/quanfoxes/parser/nodes/LabelNode.java
package fi.quanfoxes.parser.nodes;

import fi.quanfoxes.parser.Label;
import fi.quanfoxes.parser.Node;

public class LabelNode extends Node {
    private Label label;

    public LabelNode(Label label) {
        this.label = label;
    }

    public Label getLabel() {
        return label;
    }

    @Override
    public NodeType getNodeType() {
        return NodeType.LABEL_NODE;
    }
}

File: Vivid/Parser/Nodes/LabelNode.cs
using System;

public class LabelNode : Node
{
	public Label Label { get; set; }

	public LabelNode(Label label, Position? position = null)
	{
		Label = label;
		Position = position;
		Instance = NodeType.LABEL;
	}

	public override int GetHashCode()
	{
		return HashCode.Combine(Instance, Position, Label.GetName());
	}
}